flocker/node/agents/blockdevice.py

```diff
@@ -1264,6 +1381,72 @@ def shared_state_changes(self):
         )
 
 
+# Mapping from desired and discovered dataset state to
+# IStateChange factory. (The factory is expected to take
+# ``desired_dataset`` and ``discovered_dataset``.
+DATASET_TRANSITIONS = {
```

Comment:
This might be more readable if you did:
1. `Desired = DatasetStates; Current = DatasetStates`
2. define a `DO_NOTHING`
3. add comments where necessary.

E.g.

```
Desired.MOUNTED: {
    Current.ATTACHED_ELSEWHERE: DO_NOTHING,  # other node will need to detach first
    Current.ATTACHED: MountBlockDevice.from_dataset,
}
```